components/components.go

```diff
@@ -2,144 +2,133 @@
-	Name() string
-
-	// Start called upon server start.
-	// Implements component-specific poller start logic.
-	Start() error
-
-	// Returns the current states of the component.
-	States(ctx context.Context) ([]State, error)
-
-	// Returns all the events from "since".
-	Events(ctx context.Context, since time.Time) ([]Event, error)
-
-	// Called upon server close.
-	// Implements copmonent-specific poller cleanup logic.
-	Close() error
-}
-
-// HealthSettable is an optional interface that can be implemented by components
-// to allow setting the health state.
-type HealthSettable interface {
-	// SetHealthy sets the health state to healthy.
-	SetHealthy() error
-}
-
-type State struct {
-	Name      string            `json:"name,omitempty"`
-	Healthy   bool              `json:"healthy,omitempty"`
-	Health    string            `json:"health,omitempty"`     // Healthy, Degraded, Unhealthy
-	Reason    string            `json:"reason,omitempty"`     // a detailed and processed reason on why the component is not healthy
-	Error     string            `json:"error,omitempty"`      // the unprocessed error returned from the component
-	ExtraInfo map[string]string `json:"extra_info,omitempty"` // any extra information the component may want to expose
+var (
+	defaultSetMu sync.RWMutex
+	defaultSet   = make(map[string]Component)
+)
 
-	SuggestedActions *common.SuggestedActions `json:"suggested_actions,omitempty"`
+// GetAllComponents returns all the components in the default set.
+func GetAllComponents() map[string]Component {
+	defaultSetMu.RLock()
+	defer defaultSetMu.RUnlock()
+	return defaultSet
 }
 
-const (
-	StateHealthy      = "Healthy"
-	StateUnhealthy    = "Unhealthy"
-	StateInitializing = "Initializing"
-	StateDegraded     = "Degraded"
-)
+// IsComponentRegistered checks if a component is registered in the default set.
+func IsComponentRegistered(name string) bool {
+	defaultSetMu.RLock()
+	defer defaultSetMu.RUnlock()
 
-type Event struct {
-	Time             metav1.Time              `json:"time"`
-	Name             string                   `json:"name,omitempty"`
-	Type             common.EventType         `json:"type,omitempty"`
-	Message          string                   `json:"message,omitempty"`    // detailed message of the event
-	ExtraInfo        map[string]string        `json:"extra_info,omitempty"` // any extra information the component may want to expose
-	SuggestedActions *common.SuggestedActions `json:"suggested_actions,omitempty"`
+	return isComponentRegistered(defaultSet, name)
 }
 
-type Metric struct {
-	UnixSeconds         int64   `json:"unix_seconds"`
-	MetricName          string  `json:"metric_name"`
-	MetricSecondaryName string  `json:"metric_secondary_name,omitempty"`
-	Value               float64 `json:"value"`
+// isComponentRegistered checks if a component is registered in the default set.
+func isComponentRegistered(set map[string]Component, name string) bool {
+	_, ok := set[name]
+	return ok
 }
```

Comment:
Why do we need the function? Shall we delete it?

Reply:
Mainly to make these unit testable (vs. global map)